doc/src/tutorials/arch/heterogeneous_tiles.rst

```diff
@@ -0,0 +1,158 @@
+                  <site pb_type="BUFGCTRL" pin_mapping="direct"/>
+                </equivalent_sites>
+            </sub_tile>
+            <sub_tile name="BUFGCTRL_1" capacity="8">
+                <clock name="I0" num_pins="1"/>
+                <clock name="I1" num_pins="1"/>
+                <input name="CE0" num_pins="1"/>
+                <input name="CE1" num_pins="1"/>
+                <input name="IGNORE0" num_pins="1"/>
+                <input name="IGNORE1" num_pins="1"/>
+                <input name="S0" num_pins="1"/>
+                <input name="S1" num_pins="1"/>
+                <output name="O" num_pins="1"/>
+                <fc in_type="abs" in_val="2" out_type="abs" out_val="2"/>
+                <pinlocations pattern="custom">
+                    <loc side="top">BUFGCTRL_1.S1 BUFGCTRL_1.CLK_BUFG_I0 BUFGCTRL_1.CE1 BUFGCTRL_1.CLK_BUFG_I1 BUFGCTRL_1.IGNORE1 BUFGCTRL_1.IGNORE0 BUFGCTRL_1.CE0 BUFGCTRL_1.S0</loc>
+                    <loc side="right">BUFGCTRL_1.CLK_BUFG_I0 BUFGCTRL_1.CLK_BUFG_I1 BUFGCTRL_1.CLK_BUFG_O</loc>
+                </pinlocations>
+                <equivalent_sites>
+                  <site pb_type="BUFGCTRL" pin_mapping="direct"/>
+                </equivalent_sites>
+            </sub_tile>
+            <sub_tile name="BUFGCTRL_2" capacity="4">
+                <clock name="I0" num_pins="1"/>
+                <clock name="I1" num_pins="1"/>
+                <input name="CE0" num_pins="1"/>
+                <input name="CE1" num_pins="1"/>
+                <input name="IGNORE0" num_pins="1"/>
+                <input name="IGNORE1" num_pins="1"/>
+                <input name="S0" num_pins="1"/>
+                <input name="S1" num_pins="1"/>
+                <output name="O" num_pins="1"/>
+                <fc in_type="abs" in_val="2" out_type="abs" out_val="2"/>
+                <pinlocations pattern="custom">
+                    <loc side="right">BUFGCTRL_1.S1 BUFGCTRL_1.CLK_BUFG_I0 BUFGCTRL_1.CE1 BUFGCTRL_1.CLK_BUFG_I1 BUFGCTRL_1.IGNORE1 BUFGCTRL_1.IGNORE0 BUFGCTRL_1.CE0 BUFGCTRL_1.S0</loc>
+                    <loc side="left">BUFGCTRL_1.CLK_BUFG_I0 BUFGCTRL_1.CLK_BUFG_I1 BUFGCTRL_1.CLK_BUFG_O</loc>
+                </pinlocations>
+                <equivalent_sites>
+                  <site pb_type="BUFGCTRL" pin_mapping="direct"/>
+                </equivalent_sites>
+            </sub_tile>
+        </tile>
+    </tiles>
+
+    <complexblocklist>
+        <pb_type name="BUFGCTRL"/>
+            <clock name="I0" num_pins="1"/>
+            <clock name="I1" num_pins="1"/>
+            <input name="CE0" num_pins="1"/>
+            <input name="CE1" num_pins="1"/>
+            <input name="IGNORE0" num_pins="1"/>
+            <input name="IGNORE1" num_pins="1"/>
+            <input name="S0" num_pins="1"/>
+            <input name="S1" num_pins="1"/>
+            <output name="O" num_pins="1"/>
+            <mode ...>
+            ...
+        </pb_type>
+    </complexblocklist>
+
```

Comment:
I think that after giving the XML, you need to have text which walks through what it represents. So something like:

    The above `BUFG_TILE` contains three types of sub-tiles (`BUFGCTRL_0`, `BUFGCTRL_1` and `BUFGCTRL_2`).
    While each sub-tile type can contain the same pb_type (equivalent_sites of `BUFGCTRL`), the differ in two ways:
        1) Each sub-tile has different pin locations. For example `BUFGCTRL_0` has the `I1` pins on the top side of the tile, while `BUFGCTRL_1` and `BUGCTRL_2` have them on the right and left sides respectively.
        2) Each sub-tile has a different 'capacity' (i.e. a different number of sites). `BUFGCTRL_1` and `BUFGCTRL_2` have capacity 4, while `BUFGCTRL_1` has capacity 8. As a result the `BUFG_TILE` can implement a total of 16 `BUFGCTRL` blocks.

Reply:
As an aside, it may help to come up with a more distinct name for the sub-tiles in the example. `BUFGCTRL_1` sub-tile is very similar to `BUFGCTRL` pb-type. Maybe use `BUFGSUBTILE` instead for clarity?